gcore/geoheif.cpp

```diff
@@ -124,12 +124,12 @@ void GeoHEIF::extractSRS(const uint8_t *payload, size_t length) const
     }
     else if (crsEncoding == "curi")
     {
-        if ((crs.at(0) != '[') || (crs.at(crs.length() - 1) != ']'))
+        if ((crs.at(0) != '[') || (crs.at(crs.length() - 2) != ']'))
```

Comment:
this change looks wrong to me. The last character is at index length - 1, not -2.   Or isn't the closing bracket the last character?
if it is, more idiomatic would be:
```suggestion
        if ((crs.at(0) != '[') || (crs.back() != ']'))
```
or maybe there's a trailing null character... ? In which case, a code comment would be welcome

in any case, a check on crs.length() should also be added. Otherwise crs.at() might throw an exception. In the GDAL code base we tend to avoid relying on exceptions

Reply:
There is a null character to terminate the string. I've updated the logic to check that is present, which hopefully makes the code clearer. I also added a comment on the format.